tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -0,0 +1,99 @@
+"""
+Name: post_copy_request_to_queue.py
+Description:  Posts to two queues - one of the queues notifies copy_files_to_archive lambda that a file is ready for copy
+and the other queue updates the file status to STAGED in the DB.
+
+"""
+from enum import Enum
+import json
+import os
+import boto3
+from typing import Dict, Any, List, Optional
+from datetime import datetime, timezone
+import shared_recovery
+import requests_db
+import database
+from requests_db import DatabaseError
+
+PREFIX = 'dev'
+DB_HOST = 'dev-drdb-host'
+DATABASE_NAME = 'postgres'
+DATABASE_PORT = '5432'
+DATABASE_USER = 'rhassan'
+OS_ENVIRON_DB_QUEUE_URL_KEY = 'DB_QUEUE_URL'
+OS_ENVIRON_RECOVERY_QUEUE_URL_KEY = 'DB_QUEUE_URL_RECOVERY'
+
+
+def lambda_handler(event, context):
+    """
+    Lambda handler. This lambda queries all entries from orca_recoverfile table 
+    that match the given filename and whose status_id is 'PENDING'.
+    The result is then sent to the staged-recovery-queue SQS.
+    The status_id is updated to STAGED and then sent to status-update-queue SQS.
+    
+        Environment Vars:
+            DATABASE_PORT (string): the database port. The standard is 5432.
+            DATABASE_NAME (string): the name of the database.
+            DATABASE_USER (string): the name of the application user.
+    Args:
+        event:
+            A dictionary from the S3 bucket. See schemas/input.json for more information.
+        context: An object required by AWS Lambda. Unused.
+    Raises:
+        None
+    """
+    # grab the filename from event
+    for record in event['Records']:
+        filename = record['s3']['object']['key']
+
+    sql = """
+        SELECT
+            job_id, granule_id, restore_destination
+        FROM
+            orca_recoverfile
+        WHERE
+            filename = %s
```

Comment:
I will have to double check but I believe you may want to use key_path instead of filename because I believe the return schema ill return the full key path not just the filename.

Reply:
it will return the name of the object in the s3 bucket. 
For example, if the object name in S3 that was restored is "f1.doc", it can be retrieved from the json event input like this:

        for record in event['Records']:
          filename = record['s3']['object']['key']

filename will be equal to f1.doc in this case.

Reply:
Ture but does that equate to the filename in the database. If I recall, filename in the database is just the name of the file. Where you would really need the key_path out of the database to compared to the filename retrieved from the JSON record.